embassy-usb/src/class/hid.rs

```diff
@@ -369,8 +369,30 @@ impl<'d, D: Driver<'d>, const N: usize> HidReader<'d, D, N> {
     }
 }
 
+/// HID protocol
+#[repr(u8)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum Protocol {
+    /// Boot protocol, custom reports are ignored
+    Boot = 0,
+    /// Custom reports
+    Report = 1,
+}
+
 /// Handler for HID-related control requests.
 pub trait RequestHandler {
+    /// Gets the current HID protocol
+    fn get_protocol(&self) -> Option<Protocol> {
```

Comment:
None here means "return 1", it has no special meaning. I think it'd make more sense to make this method not return an Option, and make the default impl return 1.